packages/web/src/components/navbar.tsx

```diff
@@ -0,0 +1,308 @@
+import { useState } from 'react';
+import { useNavigate, NavLink, Link } from 'react-router-dom';
+import {
+  PlusIcon,
+  ChevronDownIcon,
+  SparklesIcon,
+  ImportIcon,
+  TrashIcon,
+  ShareIcon,
+  ExternalLinkIcon,
+} from 'lucide-react';
+import { TitleCellType } from '@srcbook/shared';
+
+import { SessionType } from '@/types';
+import { SrcbookLogo } from '@/components/logos';
+import { Button } from '@/components/ui/button';
+import GenerateSrcbookModal from '@/components/generate-srcbook-modal';
+import DeleteSrcbookModal from '@/components/delete-srcbook-dialog';
+import { ExportSrcbookModal, ImportSrcbookModal } from '@/components/import-export-srcbook-modal';
+import useTheme from '@/components/use-theme';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuShortcut,
+  DropdownMenuTrigger,
+} from '@/components/ui/dropdown-menu';
+import { createSession, createSrcbook } from '@/lib/server';
+
+function LightDarkModeDebugChanger() {
+  const { theme, toggleTheme } = useTheme();
+
+  if (process.env.NODE_ENV === 'production') {
+    return null;
+  }
+
+  return (
+    <div className="absolute left-1/2 -translate-x-1/2">
+      <button
+        onClick={toggleTheme}
+        className="border-none outline-none text-muted-foreground hover:text-foreground font-semibold transition-colors"
+      >
+        {theme === 'light' ? '(DEV) Dark mode' : '(DEV) Light mode'}
+      </button>
+    </div>
+  );
+}
+
+type SessionNavbarProps = {
+  session: SessionType;
+  srcbooks: Array<SessionType>;
+  title: string;
+  baseDir: string;
+};
+
+export function SessionNavbar(props: SessionNavbarProps) {
+  const navigate = useNavigate();
+
+  const [showGenSrcbookModal, setShowGenSrcbookModal] = useState(false);
+  const [showImportSrcbookModal, setShowImportSrcbookModal] = useState(false);
+  const [showDelete, setShowDelete] = useState(false);
+  const [showSave, setShowSave] = useState(false);
+
+  async function openSrcbook(path: string) {
+    const { result: srcbook } = await createSession({ path });
+    navigate(`/srcbooks/${srcbook.id}`);
+  }
+
+  async function onCreateSrcbook() {
+    const { result } = await createSrcbook({
+      path: props.baseDir,
+      name: 'Untitled',
+
+      // FIXME: copy the same language as the currently visible srcbook - should this be less
+      // implicit?
+      language: props.session.language,
+    });
+    openSrcbook(result.path);
+  }
```

Comment:
Note that for now, clicking "create srcbook" inline on the navbar will create a new srcbook with the same language as the current srcbook. I didn't see an explicit way this was handled in the designs - if there's a better way to do this that comes to mind, let me know.

Reply:
This is a good call out. I think we'll eventually probably have a modal or a blank page with a modal that asks you to select a few things (e.g., language, react or not if we choose to make that something user selects up front, possibly srcbook name, etc).